### PR TITLE
#9628: Merge third set of binary backward op from tt_eager to TTNN

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Binary Backward Issue : #9628 [Master issue : #9322 ]

### List of ops completed
- binary_ne_bw
- binary_gt_bw
- binary_ge_bw
- binary_lt_bw
- min_bw
- max_bw
- div_bw
- mul_bw
- lerp_bw

**Structure**
```
ttnn
└── cpp
    └── ttnn
        └── operations
            └── ...
            └── eltwise
                └── binary
                └── binary_backward
                    ├── device
                    │   ├── binary_backward_op.cpp
                    │   ├── binary_backward_op.hpp
                    ├── binary_backward.hpp
                    └── binary_backward_pybind.hpp
```

**Test files**
- binary_gt_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_gt.py`
- binary_ne_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_ne.py`
- binary_ge_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_ge.py`
- binary_lt_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_binary_lt.py`
- min_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_min.py`
- max_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_max.py`
- div_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_div.py`
- mul_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_mul.py`
- lerp_bw : `tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_lerp.py`

**CI :** 

- [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9722419832) - PASSED
- [[post-commit] Fast Dispatch unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9722437819) - PASSED
- [[post-commit] Slow Dispatch unit tests](https://github.com/tenstorrent/tt-metal/actions/runs/9722438194) - PASSED
- [post-commit] models tests
- Device perf regressions and output report
- Model perf regressions and output report
- Nightly fast dispatch tests

### CODE REVIEW

**Review by eyonland — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_mul.py [hunk 1]
```diff
@@ -61,23 +61,23 @@ def test_bw_mul_opt_output(input_shapes, device, are_required_outputs, pass_queu
...
-        tt_output_tensor_on_device = tt_lib.tensor.mul_bw(
+        tt_output_tensor_on_device = ttnn.mul_bw(
             grad_tensor,
             input_tensor_a,
             input_tensor_b,
             are_required_outputs=are_required_outputs,
-            input_a_grad=input_a_grad,
-            input_b_grad=input_b_grad,
+            optional_input_a_grad=input_a_grad,
+            optional_input_b_grad=input_b_grad,
```
Comment (ayerofieiev-tt) [inline tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_mul.py:70]: imho, these arguments should not have "optional" in their name. The fact that they are kw_only is enough to reflect that.
Its ok to keep it like that in this PR if it is not added in this PR, but lets make sure we address it in future changes.

File: ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp [hunk 2]
```diff
@@ -528,11 +735,24 @@ std::function<std::vector<std::optional<ttnn::Tensor>>(const Tensor&, const Tens
...
             return _binary_eq_bw_overload;
+        case BinaryBackwardOpType::MUL_BW:
+            return _mul_bw_overload;
         default:
             TT_ASSERT(false && "Undefined op type");
             return 0;
     }
 }
+
+std::function<std::vector<ttnn::Tensor>(const Tensor&, const Tensor&, const Tensor&, const Tensor&, const MemoryConfig&)> get_overload_function(BinaryBackwardOpType OpType){
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.cpp:746]: is this going to evolve? I thought it is the last part of migration

File: ttnn/ttnn/operations/binary_backward.py [hunk 3]
```diff
@@ -85,7 +85,11 @@ def _golden_function_backward_with_string(
         pyt_y.backward(gradient=grad_tensor)
         golden_tensor = [sum_result.grad, sum_result.grad]
         return golden_tensor
-    pyt_y = torch_op(input_tensor_a, input_tensor_b, value=value)
+
+    if torch_op == torch.div:
+        pyt_y = torch_op(input_tensor_a, input_tensor_b, rounding_mode=value)
+    else:
+        pyt_y = torch_op(input_tensor_a, input_tensor_b, value=value)
```
Comment (ayerofieiev-tt) [inline ttnn/ttnn/operations/binary_backward.py:92]: So we now have mode, rounding_mode and value kw_only options?

**Review by ayerofieiev-tt — APPROVED** [review]
overall lgtm

**Review by tt-aho — APPROVED** [review]

**Pull request merged**